core/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -710,6 +728,40 @@ private void maybeTrimTranslog() {
         }
     }
 
+    private void syncGlobalCheckpoints() {
+        for (final IndexShard shard : this.shards.values()) {
+            if (shard.routingEntry().active() && shard.routingEntry().primary()) {
+                switch (shard.state()) {
+                    case CLOSED:
+                    case CREATED:
+                    case RECOVERING:
+                    case RELOCATED:
+                        continue;
+                    case POST_RECOVERY:
+                        assert false : "shard " + shard.shardId() + " is in post-recovery but marked as active";
+                        continue;
+                    case STARTED:
+                        try {
+                            shard.acquirePrimaryOperationPermit(
+                                    ActionListener.wrap(
+                                            releasable -> {
+                                                try (Releasable ignored = releasable) {
+                                                        shard.maybeSyncGlobalCheckpoint("background");
+                                                }
+                                            },
+                                            e -> logger.info("failed to execute background global checkpoint sync", e)),
```

Comment:
just wondering if this could as well be an `AlreadyClosedException | IndexShardClosedException` and we should not log then.